### ISSUE TITLE
Improve documentation for the View classes

### ISSUE THREAD

**Issue by dschult** [networkx/networkx#4765]
The documentation for the edges/nodes/degree/subgraph views are not always clear (see #4745 ).
We should take a shot at improving them before v3.0....   maybe even v2.6

**Comment by salym** [comment]
I agree I remember having a hard time understanding them and especially the uncertainty to get a deep or a shallow copy.
